src/features/account/controllers/account-controller.test.ts

```diff
@@ -1,35 +1,93 @@
-import type { NextFunction, Request, Response } from 'express';
-import type { MockProxy } from 'vitest-mock-extended';
-import { mock } from 'vitest-mock-extended';
+import type { Request, Response } from 'express';
+import { mock, mockDeep } from 'vitest-mock-extended';
 
 import { AccountController } from '@/features/account/controllers/account-controller';
-import type { AccountRepository } from '@/features/account/repositories/account-repository/account-repository';
-import { DeleteUserAccountsService } from '@/features/account/services/delete-user-accounts-service';
-
-describe('Account Controller', () => {
-  let accountRepository: MockProxy<AccountRepository>;
-  let deleteAccountByIdService: DeleteUserAccountsService;
-  let accountController: AccountController;
-
-  beforeEach(() => {
-    accountRepository = mock<AccountRepository>();
-    deleteAccountByIdService = new DeleteUserAccountsService(accountRepository);
-    accountController = new AccountController(deleteAccountByIdService);
+import type { DeleteUserAccountsService } from '@/features/account/services/delete-user-accounts-service';
+import { accountDeleteBySchema } from '@/features/account/validators/account-find-by-id-schema';
+import { HttpError } from '@/shared/errors/http-error';
+import { HttpStatusCode } from '@/shared/protocols/http-client';
+
+const makeSut = () => {
+  const deleteUserAccountServiceMock = mock<DeleteUserAccountsService>({
+    execute: vi.fn(),
   });
 
-  it('should be able to delete an account by id', async () => {
-    const req = mock<Request>();
-    req.params.id = 'f19c169c-5fa2-406d-8de9-4d2c36dc6529';
+  const accountController = new AccountController(deleteUserAccountServiceMock);
+
+  const req = mockDeep<Request>();
+
+  const res = {
+    send: vi.fn(),
+
+    status: vi.fn().mockReturnThis(),
+  } as unknown as Response;
```

Comment:
`unknown` é ruim né, da pra definir como Response no mock igual no `req`
```suggestion
 const res = mock<Response>({
    send: vi.fn(),
    
    status: vi.fn().mockReturnThis(),
  });
```